junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -681,6 +682,8 @@ else if (methodPart.endsWith(")")) {
 	 * @return an {@code Optional} containing the outermost instance; never {@code null}
 	 * but potentially empty
 	 */
+	@API(status = DEPRECATED, since = "5.4")
+	@Deprecated
```

Comment:
Add pointer to the new API?

Reply:
This is an internal API. So if we no longer use it, we can delete the "dead code".

Reply:
Well, someone could mix an older version of an engine with a newer version of the Platform, e.g. when upgrading their build tool/IDE but not their engine dependency. Thus, I think we might want to keep it around for a while. Thoughts?